device/store.go

```diff
@@ -0,0 +1,166 @@
+// Copyright © 2020 Ispirata Srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package device
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/astarte-platform/astarte-go/interfaces"
+)
+
+type mqttMessage struct {
+	storageId      int
+	absoluteExpiry int64
+	retention      interfaces.AstarteMappingRetention
+	topic          string
+	qos            uint8
+	payload        []byte
+}
+
+func (d *Device) getDbDir() string {
+	// TODO change
+	dbDir := filepath.Join("/tmp/prova1/db")
+	os.MkdirAll(dbDir, 0700)
+	return dbDir
+}
+
+func initializeDb(d *Device) error {
```

Comment:
I guess we should use a migration system here, e.g.: https://github.com/golang-migrate/migrate

Reply:
Thanks, I included it in the newer revision